src/lib.rs

```diff
@@ -140,6 +146,15 @@ impl<ValueType> PartialEq for AssociativePositionalList<ValueType>
         let mut it1 = self.iter();
         let mut it2 = other.iter();
         loop {
+            match (it1.next(), it2.next()) {
+                (None, a) => {
+                    return a.is_none();
+                }
+                (Some(x), Some(y)) if x != y => {
+                    return false;
+                }
+                _ => {}
+            }
```

Comment:
oops, this bit isn't autofmt. I got keep changing things, then autofmt happened on save, and I forgot I'd changed this bit.